Tests/AppTests/AnalyzerTests.swift

```diff
@@ -180,20 +184,45 @@ class AnalyzerTests: AppTestCase {
         // add existing versions (to be reconciled)
         try Version(package: pkg,
                     commit: "commit0",
+                    // FIXME: add commitDate after cherry-pick
+                    //                    commitDate: T0,
                     latest: .defaultBranch,
                     packageName: "foo-1",
                     reference: .branch("main")).save(on: app.db).wait()
         try Version(package: pkg,
                     commit: "commit0",
+                    // FIXME: add commitDate after cherry-pick
+                    //                    commitDate: T0,
                     latest: .release,
                     packageName: "foo-1",
                     reference: .tag(1, 0, 0)).save(on: app.db).wait()
 
         Current.fileManager.fileExists = { _ in true }
-        Current.shell.run = { cmd, path in
-            if cmd.string == "git tag" {
-                return ["1.0.0", "1.1.1"].joined(separator: "\n")
+
+        Current.git.commitCount = { _ in 12 }
+        Current.git.firstCommitDate = { _ in .t0 }
+        Current.git.lastCommitDate = { _ in .t2 }
+        Current.git.getTags = { _ in [.tag(1, 0, 0), .tag(1, 1, 1)] }
+        Current.git.revisionInfo = { ref, _ in
+            // simulate the following scenario:
+            //   - main branch has moved from commit0 -> commit3 (timestamp 3)
+            //   - 1.0.0 has been re-tagged (!) from commit0 -> commit1 (timestamp 1)
+            //   - 1.1.1 has been added at commit2 (timestamp 2)
+            switch ref {
+                // NB: case .tag(1, 0, 0): does not compile, that the reason for the
+                // weird where clause (swift 5.3)
+                case _ where ref == .tag(1, 0, 0):
+                    return .init(commit: "commit1", date: .t1)
+                case _ where ref == .tag(1, 1, 1):
+                    return .init(commit: "commit2", date: .t2)
+                case .branch("main"):
+                    return .init(commit: "commit3", date: .t3)
+                default:
+                    fatalError("unexpected reference: \(ref)")
             }
+        }
```

Comment:
In other tests it's much more transparent now how the git commands are set up. Compare this to

```swift
            // New Git.revisionInfo (per ref - default branch & tags)
             // main branch has moved from commit0 -> commit3 (timestamp 3)
             // 1.0.0 has been re-tagged (!) from commit0 -> commit1 (timestamp 1)
             // 1.1.1 has been added at commit2 (timestamp 2)
             if cmd.string == #"git log -n1 --format=format:"%H-%ct" "main""# { return "commit3-3" }
             if cmd.string == #"git log -n1 --format=format:"%H-%ct" "1.0.0""# { return "commit1-1" }
             if cmd.string == #"git log -n1 --format=format:"%H-%ct" "1.1.1""# { return "commit2-2" }

             // Git.commitCount
             if cmd.string == "git rev-list --count HEAD" { return "12" }

             // Git.firstCommitDate
             if cmd.string == #"git log --max-parents=0 -n1 --format=format:"%ct""# { return "0" }

             // Git.lastCommitDate
             if cmd.string == #"git log -n1 --format=format:"%ct""# { return "2" }
```

Unless you knew how the git commands are set up it was really hard to configure the mock.